rclcpp/src/rclcpp/publisher.cpp

```diff
@@ -170,7 +170,13 @@ PublisherBase::setup_intra_process(
   StoreMessageCallbackT callback,
   const rcl_publisher_options_t & intra_process_options)
 {
-  auto intra_process_topic_name = std::string(this->get_topic_name()) + "/_intra";
+  const char * topic_name = this->get_topic_name();
+  if (topic_name) {
+    auto intra_process_topic_name = std::string(topic_name) + "/_intra";
```

Comment:
Have you tried to compile the patch locally (and ran the tests)? It look to me that the variable has a local scope and therefore isn't available below where it is being used.

Reply:
badly, it becomes a local scope within if{}, and I changed it. Now all have passed the build and tests, thanks !